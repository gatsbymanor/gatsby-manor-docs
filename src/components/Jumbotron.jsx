import React from 'react';
import Link from 'gatsby-link';
import classNames from 'classnames';


const Jumbotron = (props) => {
  const { data, center } = props;

  let ActionButton = null;

  // CallTAction present?
  if(!!data.lead.callToAction) {
    ActionButton = (
      <p className="lead">
        <Link to={data.lead.callToAction.path} color="primary">{data.lead.callToAction.text}</Link>
      </p>
    );
  }

  let jumbotronClassses = classNames({
    'jumbotron': true,
    '-center': center ? true : false,
  });


  return (
    <div className={jumbotronClassses}>
      <h1 className="title">{data.lead.title}</h1>
      <p className="lead">
        <span>{data.lead.partOne}</span>
        <br/>
        <span>{data.lead.partTwo}</span>
      </p>
      {ActionButton}
    </div>
  );
};

export default Jumbotron;
