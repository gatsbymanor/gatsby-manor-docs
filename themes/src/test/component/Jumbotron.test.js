import React from 'react';
import Jumbotron from '../../src/components/Jumbotron';

describe('<Jumbotron />', () => {
  const testData = {
    lead: {
      title: 'Theme Gallery',
      partOne: 'Developers deserve beautiful designs.',
      partTwo: 'Find a theme. Let Jay do the rest.',
      callToAction: {
        path: '/hello',
        text: 'hello world',
      },
    }
  };


  it('should render without center class', () => {
    const component = shallow(
      <Jumbotron data={testData} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render with center class', () => {
    const component = shallow(
      <Jumbotron center data={testData} />
    );

    expect(component).toMatchSnapshot();
  });
});
