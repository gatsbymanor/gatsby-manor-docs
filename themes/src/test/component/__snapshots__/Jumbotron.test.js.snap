// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> should render with center class 1`] = `
<div
  className="jumbotron -center"
>
  <h1
    className="jumbotron--title"
  >
    Theme Gallery
  </h1>
  <p
    className="jumbotron--lead"
  >
    <span>
      Developers deserve beautiful designs.
    </span>
    <br />
    <span>
      Find a theme. Let Jay do the rest.
    </span>
  </p>
  <p
    className="jumbotron--lead"
  >
    <GatsbyLink
      className="jumbotron--lead--action -primary"
      to="/hello"
    >
      hello world
    </GatsbyLink>
  </p>
</div>
`;

exports[`<Jumbotron /> should render without center class 1`] = `
<div
  className="jumbotron"
>
  <h1
    className="jumbotron--title"
  >
    Theme Gallery
  </h1>
  <p
    className="jumbotron--lead"
  >
    <span>
      Developers deserve beautiful designs.
    </span>
    <br />
    <span>
      Find a theme. Let Jay do the rest.
    </span>
  </p>
  <p
    className="jumbotron--lead"
  >
    <GatsbyLink
      className="jumbotron--lead--action -primary"
      to="/hello"
    >
      hello world
    </GatsbyLink>
  </p>
</div>
`;
